presto-main/src/test/java/com/facebook/presto/operator/aggregation/AggregationTestUtils.java

```diff
@@ -41,6 +44,11 @@ public static void assertAggregation(InternalAggregationFunction function, Objec
         assertAggregation(function, expectedValue, new Page(blocks));
     }
 
+    public static void assertAggregationWithAssertion(InternalAggregationFunction function, BiConsumer<Object, Object> equalAssertion, Object expectedValue, Block... blocks)
```

Comment:
I assume the intent is to add a function that matches `assertAggregation(InternalAggregationFunction function, Object expectedValue, Page page)` in functionality, but uses custom `assertEquals`. If that's the case, then I think it needs to look like so:

```
    public static void assertAggregation(InternalAggregationFunction function, Object expectedValue, Page page)
    {
        BiConsumer<Object, Object> equalAssertion;
        if (expectedValue instanceof Double && !expectedValue.equals(Double.NaN)) {
            equalAssertion = (actual, expected) -> assertEquals((double) actual, (double) expected, 1e-10);
        }
        else if (expectedValue instanceof Float && !expectedValue.equals(Float.NaN)) {
            equalAssertion = (actual, expected) -> assertEquals((float) actual, (float) expected, 1e-10f);
        }
        else {
            equalAssertion = (actual, expected) -> assertEquals(actual, expected);
        }

        assertAggregation(function, equalAssertion, expectedValue, page);
    }

    public static void assertAggregation(InternalAggregationFunction function, BiConsumer<Object, Object> equalAssertion, Object expectedValue, Page page)
    {
        int positions = page.getPositionCount();
        for (int i = 1; i < page.getChannelCount(); i++) {
            assertEquals(positions, page.getBlock(i).getPositionCount(), "input blocks provided are not equal in position count");
        }
        if (positions == 0) {
            assertAggregationInternal(function, equalAssertion, expectedValue, new Page[] {});
        }
        else if (positions == 1) {
            assertAggregationInternal(function, equalAssertion, expectedValue, page);
        }
        else {
            int split = positions / 2; // [0, split - 1] goes to first list of blocks; [split, positions - 1] goes to second list of blocks.
            Page page1 = page.getRegion(0, split);
            Page page2 = page.getRegion(split, positions - split);
            assertAggregationInternal(function, equalAssertion, expectedValue, page1, page2);
        }
    }
```

`assert-with-assertion` sounds strange, hence, let's keep the name `assertAggregation`. I see that because of some existing usage we can only add a version that takes `Page`, but that should be fine given how easy it is to convert a list of `Blocks` into a `Page` in the caller (`new Page(blocks)`).

Reply:
@mbasmanova Thank you, this is a great idea.  This along with your suggestion on a BiFunction below simplifies things a lot. 👍